jenkins/deploy.sh

```diff
@@ -40,24 +40,37 @@ DIST_PL=${DIST_PL:-"dist"}
 ###### Build the path of jar(s) to be deployed ######
 MVN_SETTINGS=${MVN_SETTINGS:-"jenkins/settings.xml"}
 MVN="mvn -B -Dmaven.wagon.http.retryHandler.count=3 -DretryFailedDeploymentCount=3 -s $MVN_SETTINGS"
-ART_ID=`$MVN help:evaluate -q -pl $DIST_PL -Dexpression=project.artifactId -DforceStdout`
-ART_GROUP_ID=`$MVN help:evaluate -q -pl $DIST_PL -Dexpression=project.groupId -DforceStdout`
-ART_VER=`$MVN help:evaluate -q -f $DIST_PL -Dexpression=project.version -DforceStdout`
-CUDA_CLASSIFIER=`mvn help:evaluate -q -pl $DIST_PL -Dexpression=cuda.version -DforceStdout`
+function mvnEval {
+    $MVN help:evaluate -q -DforceStdout -pl $1 -Dexpression=$2
+}
+ART_ID=$(mvnEval $DIST_PL project.artifactId)
+ART_GROUP_ID=$(mvnEval $DIST_PL project.groupId)
+ART_VER=$(mvnEval $DIST_PL project.version)
+DEFAULT_CUDA_CLASSIFIER=$(mvnEval $DIST_PL cuda.version)
+CUDA_CLASSIFIERS=${CUDA_CLASSIFIERS:-"$DEFAULT_CUDA_CLASSIFIER"}
 
 SQL_PL=${SQL_PL:-"sql-plugin"}
 POM_FILE=${POM_FILE:-"$DIST_PL/target/parallel-world/META-INF/maven/${ART_GROUP_ID}/${ART_ID}/pom.xml"}
 OUT_PATH=${OUT_PATH:-"$DIST_PL/target"}
 SIGN_TOOL=${SIGN_TOOL:-"gpg"}
 
 FPATH="$OUT_PATH/$ART_ID-$ART_VER"
-cp $FPATH-$CUDA_CLASSIFIER.jar $FPATH.jar
+DEPLOY_TYPES=''
+DEPLOY_FILES=''
+IFS=',' read -a CUDA_CLASSIFIERS_ARR <<< "$CUDA_CLASSIFIERS"
+for classifier in "${CUDA_CLASSIFIERS_ARR[@]}"; do
+    DEPLOY_TYPES="${DEPLOY_TYPES},jar"
+    DEPLOY_FILES="${DEPLOY_FILES},${FPATH}-${classifier}.jar"
+done
+# Remove the first char ','
+DEPLOY_TYPES=${DEPLOY_TYPES#*,}
+DEPLOY_FILES=${DEPLOY_FILES#*,}
```

Comment:
Maybe I'm missing something, but parsing the classifiers into an array and then trying to walk the array seems more complicated than it needs to be.  Instead I think we can simply use CUDA_CLASSIFIERS directly, preserving the commas and replacing what we need.  For example, I think the following is equivalent and doesn't require fixups at the end:
```suggestion
DEPLOY_TYPES=$(echo $CUDA_CLASSIFIERS | sed -e 's;[^,]*;jar;g')
DEPLOY_FILES=$(echo $CUDA_CLASSIFIERS | sed -e "s;\([^,]*\);${FPATH}-\1.jar;g")
```

Reply:
Good suggestion, thanks a lot!